### DISCUSSION TITLE
Saving pages in Git repository saves one locale in root directory instead of the locale dir

### DISCUSSION THREAD

**Discussion by Polda18** [requarks/wiki#7121] [Error / Bug Report] [answered]
Setting a Git repository for storage wasn't painless, but we managed. I had to create a readme file in the repository first to create the `main` branch, so the code can be stored there. However, for some reason, despite the locales being displayed on the site as it should be, in their respective directories (`en` for English, and `cs` for Czech), in the GitHub repository, the `cs` locale is saved in the root directory instead, only `en` is saved in a subdir. That makes broken links on the pages, as we have "Edit on GitHub" feature enabled, and that links to non-existent files. I tried to move the Markdown files to the correct directory, and everything seemed fine at first, but as I expanded one of the pages again, it created a new copy in the root directory again. That's not okay.

This is the system we're using:
![sysinfo](https://czghost.is-a-virg.in/6xhjeQkl4.png)

We're using (at the time of writing this bug report) the latest version 2.5.301, running on Docker Compose container under Linux (I don't know the exact Linux distro and version, may ask my friend for it, he owns the server), Node.js version is 18.19.0, and Postgres version is 11.5.

**Comment by NGPixel** [comment, marked as answer]
The site default locale is always stored at the root, while all other locales are stored in subfolders.

Feel free to submit a feature request to have the option to change this behavior:
https://requarks.canny.io/wiki

**Comment by Polda18** [comment]
Right. As per request, I've submitted a suggestion: https://requarks.canny.io/wiki/p/save-pages-always-in-a-locale-dir-instead-of-root-in-git-repositories

Did you seriously not once considered that we might want to change the default language at some point? Doing that with this "feature" creates a lot of issues. Not to mention, that this "feature" actually DOES create a bug that's related to this, and such that the GitHub link from the wiki leads to an invalid resource, as the wiki thinks the file should be in the locale directory, but instead it resides in the root. This inconsistency can lead to pretty massive confusion from people wanting to contribute, unaware of this strange quirk.

**Reply by NGPixel** [reply]
It's very easy to complain about implementation when you don't know the full context. In reality it's not that simple and there're a lot of use cases to consider. For example:

- Most users use a single locale and have a storage module enabled. What if you enable multi-locales afterwards when you already have content? New content to the default locale would start being added in a subfolder while older content would stay in the root path.

- What if you have content of your default locale both in a subdirectory and content at the root? What happens during sync then?

- If you change the default locale, do you move everything in a subfolder at that point? What if you wrote all your content in French but your default language was English, and then decide to change it?

The current implementation was done to have the least amount of changes when multi-locale is enabled.

So while you might think it's quick and obvious to implement a feature a certain way, it's not. All these use cases must be handled **and that takes time**. The way you configure your wiki is not the same as everyone else and assuming it's the case is being ignorant. There're thousands of bugs and improvements that need to be fixed/implemented and what's critical to you is completely pointless to someone else, so please keep that in mind when posting these kind of comments.

**Reply by Polda18** [reply]
You can't be serious. I had to create a readme file in order to even be able to upload to GitHub, that's obviously in the root dir. But now it actually mixes in with the images and default locale, it's a mess. If users use a single locale, saving it in the locale dir on GitHub doesn't actually hurt that much, it's just there, and if the wiki decides to add another language, at least it's prepared. Why make things easily if you can take the complicated path, right? What you basically said isn't a reason why it's done this way. It only complicates things, and for nothing. These special use cases are basically just there because someone somehow forgot about them, and you are now making excuses. I don't buy that, I call the bluff. It's a bug, you won't convince me otherwise.

**Reply by NGPixel** [reply]
Most users **DO NOT** want to have their files stored in a subfolder if they are only using a single language. Again, you think your use case applies to everyone, which is simply not true.

I never said it wasn't a bug. I explained **WHY** it was implemented this way (since you couldn't possibly comprehend why) and that some use cases still needed to be handled. Clearly you haven't understood my response.

If you're just going to complain about why things aren't done YOUR way from the start, then please go use another wiki software or build your own.
